scipy/misc/common.py

```diff
@@ -416,3 +419,249 @@ def face(gray=False):
     if gray is True:
         face = (0.21 * face[:,:,0] + 0.71 * face[:,:,1] + 0.07 * face[:,:,2]).astype('uint8')
     return face
+
+
+def inversefunc(f,
+                domain=None,
+                image=None,
+                open_domain=None,
+                accuracy=2):
+    r"""Obtain the inverse of a function.
+
+    Returns a callable that calculates the numerical inverse of the function
+    `f`. In order for the numerical inverse to exist in its domain, the
+    input function must have, continuous, strictly monotonic behavior i.e. be a
+    purely decreasing or purely increasing in that domain. By default the
+    domain interval spans all the real numbers, however it can be restricted
+    with the `domain` and `open_domain` arguments. The image of the function
+    in the interval may be provided, for cases where the function is non
+    non continuous right at the end of an open interval.
+
+    Parameters
+    ----------
+    f : callable
+        Callable representing the function to be inverted, able to take a
+        ndarray or an scalar and return an object of the same kind with the
```

Comment:
taking only a single parameter may be a bit annoying, requires users to use ``functools.partial`` or define a wrapper function for a lot of functions. One common pattern to solve that would be to add a ``args=()`` keyword, see for example ``integrate.quad``.